index.js

```diff
@@ -0,0 +1,88 @@
+const tmp = require('tmp');
+const ECS_CONTAINER_NAME = core.getInput('ecs-container-name');
+const REGION = core.getInput('aws-region');
+const SERVICE_NAME = core.getInput('service-name');
+const PROVIDED_IMAGE_TAG = core.getInput('provided-image-tag');
+const ECS_TASK_DEFINITION = core.getInput('ecs-task-definition');
+const DOCKER_TAG = core.getInput('docker-tag');
+const ssm = new aws.SSM({region: REGION});
+
+async function getTaskDefinition(ecsTaskDefinition, region) {
+    let ecs = new aws.ECS({region: region});
+
+    let output = await ecs.describeTaskDefinition({ 
+        taskDefinition: ecsTaskDefinition
+    }).promise();
+
+    return output.taskDefinition;
+}
+
+async function getParameters(path, params = [], nextToken) {
+    return ssm
+        .getParametersByPath({ Path: path, NextToken: nextToken, MaxResults: 10 })
+        .promise()
+        .then(({ Parameters, NextToken }) => {
+            const newParams = params.concat(Parameters);
+            return NextToken ? getParameters(path, newParams, NextToken) : newParams;
+        });
+}
+
+async function formatParameters(params) {
+    return params.map(x => x.ARN).map(function (arn) {
+        return { 'valueFrom': arn, 'name': arn.split('/').at(-1) }
+    });
+}
+
+async function updateTaskDefinition(appName, taskDefinition, parameters) {    
+    let selectedDefinitions = taskDefinition.containerDefinitions.filter(function(container) { 
+        return container.name === appName; 
+    });
+
+    if (selectedDefinitions.length > 1) {
+        throw new Error(`Multiple task definition containers found with name: ${appName}`);
+    }
+
+    let selectedDefinition = selectedDefinitions[0];
+
+    if (!selectedDefinition) {
+        throw new Error(`Unable to find task definition container with name: ${appName}`);
+    }
+
+    selectedDefinition.secrets = parameters
+    
+    return taskDefinition;
+}
+
+async function writeNewTaskDefinitionTmpFile(taskDefinition) {
+    let updatedTaskDefFile = tmp.fileSync({
+      prefix: 'task-definition',
+      postfix: '.json',
+      keep: true,
+      discardDescriptor: true
+    });
+
+    const newTaskDefContents = JSON.stringify(taskDefinition, null, 2);
+    fs.writeFileSync(updatedTaskDefFile.name, newTaskDefContents);
+
+    return updatedTaskDefFile.name;
+}
+
+async function run() {
+    try {
+        const taskDefinition = await getTaskDefinition(ECS_TASK_DEFINITION, REGION);
+        const parameters = await getParameters(`/ecs/${SERVICE_NAME}`);
+        const parsedParameters = await formatParameters(parameters);
+        const newTaskDefinition = await updateTaskDefinition(ECS_CONTAINER_NAME, taskDefinition, parsedParameters);
+        const newTaskDefinitionTmpFileName = await writeNewTaskDefinitionTmpFile(newTaskDefinition);
+
+        core.setOutput('task-definition', newTaskDefinitionTmpFileName);
+        core.setOutput('image-tag', PROVIDED_IMAGE_TAG == '' ? DOCKER_TAG : PROVIDED_IMAGE_TAG);
```

Comment:
```suggestion
        core.setOutput('image-tag', PROVIDED_IMAGE_TAG == '' ? DOCKER_TAG : PROVIDED_IMAGE_TAG);
```

Also occurs to me that this code is essentially asking for both `DOCKER_TAG` and the user's override Docker tag.  But we can determine the value for `DOCKER_TAG`, right?  We can decide it is (e.g.) `buildnum` or `sha` or whatever.

Reply:
Ah, nope, i realise the Docker build will have happened _before_ this step, so we can't centralise that logic here.  NVM

Reply:
It's a pity that logic is pretty loosely coupled this way 🤔